### ISSUE TITLE
Check failed: mdb_status == 0 (2 vs. 0) No such file or directory

### ISSUE THREAD

**Issue by p4tr1ckc4rs0n** [BVLC/caffe#2780]
Upon running ./train_caffenet.sh; creating the training layer data, setting up the data, loading the training mean.binaryproto file and opening the training lmdb all works.

However when it comes to loading the test mean.binaryproto file for the test data I get the following error:

Loading mean file from: /home/pwhc/caffe/Learn/imagenet_mean_test.binaryproto
F0716 13:12:13.917732  3385 db.hpp:109] Check failed: mdb_status == 0 (2 vs. 0) No such file or directory
*** Check failure stack trace: ***
    @     0x7f8337946daa  (unknown)
    @     0x7f8337946ce4  (unknown)
    @     0x7f83379466e6  (unknown)
    @     0x7f8337949687  (unknown)
    @     0x7f8337cbf5be  caffe::db::LMDB::Open()
    @     0x7f8337d16b82  caffe::DataLayer<>::DataLayerSetUp()
    @     0x7f8337d806f9  caffe::BasePrefetchingDataLayer<>::LayerSetUp()
    @     0x7f8337ca3db3  caffe::Net<>::Init()
    @     0x7f8337ca5b22  caffe::Net<>::Net()
    @     0x7f8337cb0a24  caffe::Solver<>::InitTestNets()
    @     0x7f8337cb111b  caffe::Solver<>::Init()
    @     0x7f8337cb12e6  caffe::Solver<>::Solver()
    @           0x40c4c0  caffe::GetSolver<>()
    @           0x406503  train()
    @           0x404ab1  main
    @     0x7f8336e58ec5  (unknown)
    @           0x40505d  (unknown)
    @              (nil)  (unknown)
Aborted (core dumped)

I modified the train_val.prototxt to point to the appropriate mean.binaryproto files (using absolute paths) and have checked and double checked to make sure  the paths match up

Any thoughts would be greatly appreciated!

**Comment by atique81** [comment]
@p4tr1ckc4rs0n, I am also suffering from the same problem....anyone please advise..

**Comment by lukeyeager** [comment]
`data_param.source` should point to an LMDB (see [here](https://github.com/BVLC/caffe/blob/d37f766f51d3f84b76343ac9bfdb11300d65162c/models/bvlc_alexnet/train_val.prototxt#L15-L19)).
`transform_param.mean_file` should point to a .binaryproto file (see [here](https://github.com/BVLC/caffe/blob/d37f766f51d3f84b76343ac9bfdb11300d65162c/models/bvlc_alexnet/train_val.prototxt#L10-L14)).

It sounds like you've put in the path to a .binaryproto where you need the path to an LMDB.

**Comment by p4tr1ckc4rs0n** [comment]
@lukeyeager the only other thing i changed in the `train_val.prototxt` was the **num_output: 2**  since i only want a 0/1 classification.

Data layers for caffe model:

name: "CaffeNet"
layer {
  name: "data"
  type: "Data"
  top: "data"
  top: "label"
  include {
    phase: TRAIN
  }
  transform_param {
    mirror: true
    crop_size: 227
    mean_file: "../imagenet_mean_train.binaryproto"
  }
  
  data_param {
    source: "/home/pwhc/caffe/GPRLearn/lmdb/GPR_train_lmdb"
    batch_size: 1
    backend: LMDB
  }
}
layer {
  name: "data"
  type: "Data"
  top: "data"
  top: "label"
  include {
    phase: TEST
  }
  transform_param {
    mirror: false
    crop_size: 227
    mean_file: "../imagenet_mean_test.binaryproto"
  }

  data_param {
    source: "/home/pwhc/caffe/GPRLearn/GPR_test_lmdb"
    batch_size: 1
    backend: LMDB
  }
}

**Comment by lukeyeager** [comment]
Ok I guess the log message was just confusing me. Nevermind.

Can you check that `data.mdb` exists in that directory:
```bash
$ ls /home/pwhc/caffe/GPRLearn/lmdb/GPR_train_lmdb/
data.mdb lock.mdb
```

**Comment by longjon** [comment]
Closing as this looks like a usage issue; it's very likely that a path to one of your LMDBs is not correct (e.g., are your train and test LMDBs really in different directories?).

Please ask usage questions on [caffe-users](https://groups.google.com/forum/#!forum/caffe-users). If you do think there is a bug in Caffe, you're welcome to open a new issue according to https://github.com/BVLC/caffe/wiki/Reporting-Bugs-and-Other-Issues. Thanks!

**Comment by atique81** [comment]
Dear @longjon  
I am running a caffe PR for fully-convolutional semantic segmentation made available by you (https://github.com/longjon/caffe/tree/future). I am stuck with this very problem while trying to read the lmdb files for the ground truth labels, though the image lbdb files are being read successfully. I have double-checked to make it sure that the lmdb paths are correct. I am using this python script #1698 to create my lmdb files both for images and their ground-truth labels on pascal voc dataset. 

I would highly appreciate if you kindly respond.

**Comment by tangjunfa** [comment]
I have suffer from a question like this:

E:\tang\caffe>"bin/caffe.exe" train --solver=examples/mnist/lenet_solver.prototx
t
I0407 11:40:59.773578  7044 caffe.cpp:185] Using GPUs 0
I0407 11:40:59.831086  7044 caffe.cpp:190] GPU 0: GeForce GT 730
I0407 11:41:00.128123  7044 common.cpp:31] System entropy source not available,
using fallback algorithm to generate seed instead.
I0407 11:41:00.128623  7044 solver.cpp:48] Initializing solver from parameters:

test_iter: 100
test_interval: 500
base_lr: 0.01
display: 100
max_iter: 10000
lr_policy: "inv"
gamma: 0.0001
power: 0.75
momentum: 0.9
weight_decay: 0.0005
snapshot: 5000
snapshot_prefix: "examples/mnist/lenet"
solver_mode: GPU
device_id: 0
net: "examples/mnist/lenet_train_test.prototxt"
I0407 11:41:00.130125  7044 solver.cpp:91] Creating training net from net file:
examples/mnist/lenet_train_test.prototxt
I0407 11:41:00.133124  7044 net.cpp:313] The NetState phase (0) differed from th
e phase (1) specified by a rule in layer mnist
I0407 11:41:00.133625  7044 net.cpp:313] The NetState phase (0) differed from th
e phase (1) specified by a rule in layer accuracy
I0407 11:41:00.135124  7044 net.cpp:49] Initializing net from parameters:
name: "LeNet"
state {
  phase: TRAIN
}
layer {
  name: "mnist"
  type: "Data"
  top: "data"
  top: "label"
  include {
    phase: TRAIN
  }
  transform_param {
    scale: 0.00390625
  }
  data_param {
    source: "examples/mnist/mnist_train_lmdb"
    batch_size: 64
    backend: LMDB
  }
}
layer {
  name: "conv1"
  type: "Convolution"
  bottom: "data"
  top: "conv1"
  param {
    lr_mult: 1
  }
  param {
    lr_mult: 2
  }
  convolution_param {
    num_output: 20
    kernel_size: 5
    stride: 1
    weight_filler {
      type: "xavier"
    }
    bias_filler {
      type: "constant"
    }
  }
}
layer {
  name: "pool1"
  type: "Pooling"
  bottom: "conv1"
  top: "pool1"
  pooling_param {
    pool: MAX
    kernel_size: 2
    stride: 2
  }
}
layer {
  name: "conv2"
  type: "Convolution"
  bottom: "pool1"
  top: "conv2"
  param {
    lr_mult: 1
  }
  param {
    lr_mult: 2
  }
  convolution_param {
    num_output: 50
    kernel_size: 5
    stride: 1
    weight_filler {
      type: "xavier"
    }
    bias_filler {
      type: "constant"
    }
  }
}
layer {
  name: "pool2"
  type: "Pooling"
  bottom: "conv2"
  top: "pool2"
  pooling_param {
    pool: MAX
    kernel_size: 2
    stride: 2
  }
}
layer {
  name: "ip1"
  type: "InnerProduct"
  bottom: "pool2"
  top: "ip1"
  param {
    lr_mult: 1
  }
  param {
    lr_mult: 2
  }
  inner_product_param {
    num_output: 500
    weight_filler {
      type: "xavier"
    }
    bias_filler {
      type: "constant"
    }
  }
}
layer {
  name: "relu1"
  type: "ReLU"
  bottom: "ip1"
  top: "ip1"
}
layer {
  name: "ip2"
  type: "InnerProduct"
  bottom: "ip1"
  top: "ip2"
  param {
    lr_mult: 1
  }
  param {
    lr_mult: 2
  }
  inner_product_param {
    num_output: 10
    weight_filler {
      type: "xavier"
    }
    bias_filler {
      type: "constant"
    }
  }
}
layer {
  name: "loss"
  type: "SoftmaxWithLoss"
  bottom: "ip2"
  bottom: "label"
  top: "loss"
}
I0407 11:41:00.147126  7044 layer_factory.hpp:77] Creating layer mnist
I0407 11:41:00.148126  7044 common.cpp:31] System entropy source not available,
using fallback algorithm to generate seed instead.
I0407 11:41:00.148627  7044 net.cpp:91] Creating Layer mnist
I0407 11:41:00.149127  7044 net.cpp:399] mnist -> data
I0407 11:41:00.149626  4024 common.cpp:31] System entropy source not available,
using fallback algorithm to generate seed instead.
I0407 11:41:00.149626  7044 net.cpp:399] mnist -> label
F0407 11:41:00.150126  4024 db_lmdb.hpp:14] Check failed: mdb_status == 0 (3 vs.
 0)
*** Check failure stack trace: ***

I will appreciate if you could respond.

**Comment by tarunsharma1** [comment]
I had the same error but then I realised that I did not have access to the lmdb folder (try cd into the lmdb folder). By changing permissions I got rid of the error

**Comment by linrio** [comment]
@lukeyeager  **cifar10_full_train_test.prototxt** : `name: "CIFAR10_full"
layer {
  name: "cifar"
  type: "Data"
  top: "data"
  top: "label"
  include {
    phase: TRAIN
  }
  transform_param {
    mean_file: "examples/cifar10/mean.binaryproto"
  }
  data_param {
    source: "examples/cifar10/cifar10_train_lmdb"
    batch_size: 100
    backend: LMDB
  }
}`
But I do not find  **cifar10_train_lmdb** and **mean.binaryproto** in the cifar10 file. I happen the same problem:
**Check failed: mdb_status == 0 (2 vs. 0) No such file or directory**

**Comment by wakemeupJ** [comment]
I have the same problem. Then I notice the path to the LMDB is relative path, and the system can't find it.

**Comment by RyanCV** [comment]
@wakemeupJ what do you mean it's a relative path? I use the absolute path, but it still gives the same error 
`Check failed: datum_channels > 0 (0 vs. 0) `. Could you give me some help? thanks.

**Comment by JohnTian** [comment]
I solved this issue by changing normal user to root, `$ ` to `#` in command prompt.

**Comment by yycer** [comment]
try to convert data into lmdb

**Comment by kaiseu** [comment]
it turns out to be that lmdb file path should be absolute path or execute the scripts in caffe root.

**Comment by Tahirhan** [comment]
check your path name in make_imagenet_mean.sh , there might be typing error .

**Comment by youngolutosin** [comment]
I found a solution to this. Kindly read the readme.md file in directory of the train and test data e.g. cifar10. But firstly, caffe must be compiled successfully because a build file will be generated after a successful compilation which will be used as a path when running the command specified in the readme file.
